src/statisticalmodel.jl

```diff
@@ -0,0 +1,243 @@
+    loglikelihood(model::StatisticalModel)
+
+Return the log-likelihood of the model.
+"""
+function loglikelihood end
+
+"""
+    loglikelihood(model::StatisticalModel)
+
+Return the log-likelihood of the null model corresponding to `model`.
+This is usually the model containing only the intercept.
+"""
+function nullloglikelihood end
+
+"""
+    loglikelihood(model::StatisticalModel, ::Colon)
+
+Return a vector of each observation's contribution to the log-likelihood of the model.
+In other words, this is the vector of the pointwise log-likelihood contributions.
+
+In general, `sum(loglikehood(model, :)) == loglikelihood(model)`.
+"""
+function loglikelihood end
+
+"""
+    loglikelihood(model::StatisticalModel, observation)
+
+Return the contribution of `observation` to the log-likelihood of `model`.
+"""
+function loglikelihood end
+
+"""
+    score(model::StatisticalModel)
+
+Return the score of the model, that is the gradient of the
+log-likelihood with respect to the coefficients.
+"""
+function score end
+
+"""
+    nobs(model::StatisticalModel)
+
+Return the number of independent observations on which the model was fitted. Be careful
+when using this information, as the definition of an independent observation may vary
+depending on the model, on the format used to pass the data, on the sampling plan
+(if specified), etc.
+"""
+function nobs end
+
+"""
+    dof(model::StatisticalModel)
+
+Return the number of degrees of freedom consumed in the model, including
+when applicable the intercept and the distribution's dispersion parameter.
+"""
+function dof end
+
+"""
+    mss(model::StatisticalModel)
+
+Return the model sum of squares.
+"""
+function mss end
+
+"""
+    rss(model::StatisticalModel)
+
+Return the residual sum of squares of the model.
+"""
+function rss end
+
+"""
+    informationmatrix(model::StatisticalModel; expected::Bool = true)
+
+Return the information matrix of the model. By default the Fisher information matrix
+is returned, while the observed information matrix can be requested with `expected = false`.
+"""
+function informationmatrix end
+
+"""
+    stderror(model::StatisticalModel)
+
+Return the standard errors for the coefficients of the model.
+"""
+function stderror end
+
+"""
+    vcov(model::StatisticalModel)
+
+Return the variance-covariance matrix for the coefficients of the model.
+"""
+function vcov end
+
+"""
+    weights(model::StatisticalModel)
+
+Return the weights used in the model.
+"""
+function weights end
+
+"""
+    isfitted(model::StatisticalModel)
+
+Indicate whether the model has been fitted.
+"""
+function isfitted end
+
+"""
+Fit a statistical model.
+"""
+function fit end
+
+"""
+Fit a statistical model in-place.
+"""
+function fit! end
+
+"""
+    aic(model::StatisticalModel)
+
+Akaike's Information Criterion, defined as ``-2 \\log L + 2k``, with ``L`` the likelihood
+of the model, and `k` its number of consumed degrees of freedom
+(as returned by [`dof`](@ref)).
+"""
+function aic end
```

Comment:
```suggestion
```
There's the concrete definition right below

Reply:
I normally use empty stubs because for interface docs as its easier in Documenter, and consistent. But its not really important, I've deleted them.